jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -68,13 +69,15 @@
 	@Nonnull private final ConsistencyLevel writeConsistency;
 	@Nonnull private final String retentionPolicy;
 	@Nonnull private final ImmutableMap<String,String> tags;
-	@Nonnull ImmutableList<String> typeNames;
-
 	/**
 	 * The {@link ImmutableSet} of {@link ResultAttribute} attributes of
 	 * {@link Result} that will be written as {@link Point} tags
 	 */
-	private final ImmutableSet<ResultAttribute> resultAttributesToWriteAsTags;
+	@Nonnull private final ImmutableSet<ResultAttribute> resultAttributesToWriteAsTags;
```

Comment:
tags, resultAttributesToWriteAsTags, typeNamesAsTags, attributesAsTags...
I wonder whether it would be cleaner to have a `List<TagSupplier>`, TagSupplier being a poorly named functional interface to extract tags from server/query/result.

Reply:
I'll see if I can do something like that.